bundles/org.openhab.binding.folderwatcher/src/main/java/org/openhab/binding/folderwatcher/internal/handler/FolderLocalWatcherHandler.java

```diff
@@ -0,0 +1,141 @@
+        try {
+            previousLocalListing = WatcherCommon.initStorage(currentLocalListingFile);
+        } catch (IOException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, e.getMessage());
+            logger.debug("Can't write file {}: {}", currentLocalListingFile, e.getMessage());
+            return;
+        }
+
+        if (config.pollIntervalLocal > 0) {
+            updateStatus(ThingStatus.ONLINE);
+            executionJob = scheduler.scheduleWithFixedDelay(this::refreshFolderInformation, config.pollIntervalLocal,
+                    config.pollIntervalLocal, TimeUnit.SECONDS);
+        } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "Polling interval can't be null or negative");
+        }
+
+    }
+
+    @Override
+    public void dispose() {
+        if (executionJob != null) {
+            executionJob.cancel(true);
+        }
+    }
+
+    private List<String> listFiles(String path, boolean listHidden, boolean recursive) throws IOException {
+        List<String> dirList = new ArrayList<>();
+        final File dir = new File(path);
+        Optional.ofNullable(dir.listFiles(filter -> listHidden || !filter.isHidden())).map(Arrays::stream)
+                .orElseGet(Stream::empty).forEach(file -> {
+                    if (file.isDirectory() && recursive) {
+                        try {
+                            dirList.addAll(listFiles(file.getAbsolutePath(), listHidden, recursive));
+                        } catch (IOException e) {
+                            logger.debug("Can't read directory: {}", file.getAbsolutePath(), e);
+                        }
+                    } else if (file.isFile()) {
+                        dirList.add(file.getAbsolutePath());
+                    }
+                });
+        return dirList;
+    }
+
+    private void refreshFolderInformation() {
+
+        String RootDir = new String();
+
+        if (config.localDir != null) {
+            RootDir = config.localDir;
+        } else {
+            logger.debug("Local directory is not specified");
+            return;
+        }
+
+        try {
+            currentLocalListing.clear();
+            currentLocalListing.addAll(listFiles(RootDir, config.listHiddenLocal, config.listRecursiveLocal));
+            List<String> diffLocalListing = new ArrayList<>(currentLocalListing);
+            diffLocalListing.removeAll(previousLocalListing);
+            diffLocalListing.forEach(file -> triggerChannel(CHANNEL_LOCALFILENAME, file));
+            if (!diffLocalListing.isEmpty()) {
+                WatcherCommon.saveNewListing(diffLocalListing, currentLocalListingFile);
+            }
+            previousLocalListing = new ArrayList<>(currentLocalListing);
+        } catch (IOException e) {
+            logger.debug("File manipulation error: {}", e.getMessage());
```

Comment:
Shouldn't this cause the thing to go offline?

Reply:
Same comment here: I was thinking about it. In case writing to file is failed on the next iteration channel will receive same list of new files, so it's not entirely breaking thing, but messing with duplicated channel update. Is it enough to consider that thing should go offline?